File: boj/11728.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int arr[] = new int[2000003];
    static int arr1[] = new int[1000003];
    static int arr2[] = new int[1000003];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; ++i) arr1[i] = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < m; ++i) arr2[i] = Integer.parseInt(st.nextToken());
        int i = 0, j = 0, k = 0;
        while (i < n && j < m) {
            if (arr1[i] < arr2[j]) arr[k++] = arr1[i++];
            else arr[k++] = arr2[j++];
        }
        while (i < n) arr[k++] = arr1[i++];
        while (j < m) arr[k++] = arr2[j++];
        StringBuilder sb = new StringBuilder();
        for (i = 0; i < n + m; ++i) sb.append(arr[i]).append(" ");
        sb.append("\n");
        System.out.println(sb);
    }
}

File: boj/11728.cpp
#include <iostream>

using namespace std;

int arr[2000003], arr1[1000003], arr2[1000003];

int main(void) {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);
	int n, m; cin >> n >> m;
	for (int i = 0; i < n; ++i) cin >> arr1[i];
	for (int i = 0; i < m; ++i) cin >> arr2[i];
	int i = 0, j = 0, k = 0;
	while (i < n && j < m) {
		if (arr1[i] < arr2[j]) arr[k++] = arr1[i++];
		else arr[k++] = arr2[j++];
	}
	while (i < n) arr[k++] = arr1[i++];
	while (j < m) arr[k++] = arr2[j++];
	for (int i = 0; i < n + m; ++i) cout << arr[i] << " ";
	cout << "\n";
	return 0;
}